### PR TITLE
Add initial support to Meadow F7Micro board

### PR DESCRIPTION
**Pull request by duduita**
## Summary
Add support to Wilderness Labs F7Micro board.

## Impact
Users will be able to use this board in the NuttX mainline.

## Testing
-

### CODE REVIEW

File: boards/arm/stm32f7/stm32f777zit6-meadow/kernel/Makefile [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+             $(MONO_DIR)/mini/.libs/libmono-ee-interp.a \
+             $(MONO_DIR)/utils/.libs/libmonoutils.a \
+             $(MONO_DIR)/dis/libmonodis.a \
+             $(MONO_DIR)/eglib/.libs/libeglib.a \
+             $(MONO_DIR)/metadata/.libs/libmonoruntime-config.a \
+             $(MONO_DIR)/metadata/.libs/libmonoruntimesgen.a \
+             $(MONO_DIR)/metadata/.libs/libmono-system-native.a
+
+MONO_LIBM = /usr/local/Caskroom/gcc-arm-embedded/7-2017-q4-major/gcc-arm-none-eabi-7-2017-q4-major/arm-none-eabi/lib/hard/libm.a
+MONO_LIBM=
```
Comment (acassis) [inline boards/arm/stm32f7/stm32f777zit6-meadow/kernel/Makefile:71]: Please remove

File: boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_boot.c [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      {
+        syslog(LOG_ERR, "ERROR: sam_qspi_initialize muiled\n");
+        return;
+      }
+
+    mtd = s25fl5_initialize(qspi, true);
+    if (!mtd)
+      {
+        syslog(LOG_ERR, "ERROR: s25fl5_initialize failed\n");
+        //return;
```
Comment (acassis) [inline boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_boot.c:149]: remove C++ comment

File: boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_boot.c [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+      }
+
+    ret = ftl_initialize(0, mtd);
+    if (ret < 0)
+      {
+        ferr("ERROR: Initialize the FTL layer\n");
+        //return ret;
```
Comment (acassis) [inline boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_boot.c:156]: remove C++ comment

File: boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_boot.c [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        return;
+      }
+
+    ret = mount(NULL, "/mnt/meadow0", "nxffs", 0, NULL);
+    if (ret < 0)
+      {
+        ferr("ERROR: Failed to mount the NXFFS volume: %d\n", errno);
+        return;
+      }
+*/
```
Comment (acassis) [inline boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_boot.c:172]: remove

File: boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_boot.c [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        printf ("\nread back -- dumping first 16 bytes\n");
+        for (i = 0; i < 16; i++) {
+          printf ("%02x ", buffer[i]);
+        }
+        printf("\n");
+
+        //mkfatfs /dev/mtdblock0
+        //mkfatfs("/dev/mtdblock0", &fmt);
+      }
+#endif
```
Comment (acassis) [inline boards/arm/stm32f7/stm32f777zit6-meadow/src/stm32_boot.c:208]: remove

**Comment by xiaoxiang781216** [comment]
@duduita please squash your change into one patch by `git rebase --interactive HEAD~6` and change all pick to squash except the first one.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**